src/Farmer/Arm/Web.fs

```diff
@@ -258,4 +269,37 @@ module SiteExtensions =
         interface IArmResource with
             member this.ResourceId = siteExtensions.resourceId(this.SiteName/this.Name)
             member this.JsonModel =
-                siteExtensions.Create(this.SiteName/this.Name, this.Location, [ sites.resourceId this.SiteName ]) :> _
\ No newline at end of file
+                siteExtensions.Create(this.SiteName/this.Name, this.Location, [ sites.resourceId this.SiteName ]) :> _
+
+type Slot =
+    { 
+      SlotName: string
+      Location : Location
+      ServicePlan: ResourceId
+      Site: ResourceId
+      Tags: Map<string,string>
+      AppSettings: Map<string,Setting>
+      ConnectionStrings: Map<string,Setting*ConnectionStringKind>
+    }
+    member this.ResourceName = this.Site.Name / this.SlotName
+    interface IParameters with
+        member this.SecureParameters =
+            Map.toList this.AppSettings
```

Comment:
If using `|>` syntax for this expression, put `this.AppSettings` as the start of the expression and `|>` to `Map.toList`.